src/python/phenix_apps/apps/scorch/kafka/kafka_component.py

```diff
@@ -0,0 +1,114 @@
+import subprocess
+import shlex
+import json
+import itertools
+import threading
+import time
+import sys
+import csv
+import re
+import os
+import time
+from phenix_apps.apps.scorch import ComponentBase
+from phenix_apps.common import logger, utils
+from kafka import KafkaConsumer
+from kafka.errors import KafkaError
+
+class Kafka(ComponentBase):
+    def __init__(self):
+        ComponentBase.__init__(self, 'kafka')
+        self.pid_file = f"/tmp/phenix-scorch-kafka-{self.exp_name}.pid"
+        self.started = False
+        self.configured = False
+        self.execute_stage()
+    
+    #configure just calls start
+    def configure(self):
+        #if we call the start and configure changes, only run the component once
+        if self.started:
+            return
+        else:
+            start()
+            self.configured = True
+
+    def start(self):
+        #if we've already configured the element, return
+        if self.configured:
+            return
+        self.started = True
+        self.scorch_kafka_running = True
+        logger.log('INFO', f'Starting user component: {self.name}')
+
+        #get all variables from tags
+        kafka_ips = self.metadata.get("kafka_ips", ["127.0.0.1:9092"])
+        topics = self.metadata.get("topics", [])
+        csvBool = self.metadata.get("csv", True) #if false output a JSON
+
+        #get and output the output directory to the logger
+        output_dir = self.base_dir
+        logger.log('INFO', f'Output Directory: {output_dir}')
+        os.makedirs(output_dir, exist_ok=True)
+        if csvBool:
+            self.path = os.path.join(output_dir, f'{self.name}_output.csv')
+        else:
+            self.path = os.path.join(output_dir, f'{self.name}_output.json')
+
+        kafka_ips_str = ",".join(kafka_ips)
+        topics_str = json.dumps(topics)
+
+        #pass the inputs to the python file (which we execute as a separate process)
+        executable  = "/usr/local/lib/python3.12/dist-packages/phenix_apps/apps/scorch/kafka/kafka_listener.py"
+        arguments = f"python3 {executable} {csvBool} '{self.path}' {kafka_ips_str} '{topics_str}'"
+        command = shlex.split(arguments)
+
+        try:
+            log_file = open("/var/log/phenix/phenix.log", '+a')
+            response = subprocess.Popen(command, stdin=subprocess.DEVNULL, stdout=log_file, stderr=log_file, start_new_session=True)
+            self._create_pid_file(response.pid) #write PID to a file so that it can be found and killed later
+            response.poll() #prevents hang
+        except Exception as e:
+            self.eprint("ERROR", f"Error running listener executable. See: {e}")
```

Comment:
eprint only takes one argument, the message to print, remove "ERROR" as first argument.